cmake/ProjectPaillier.cmake

```diff
@@ -0,0 +1,31 @@
+include(ExternalProject)
+set(PAILLIER_LIB_SUFFIX a)
+
+ExternalProject_Add(paillier
+    PREFIX ${CMAKE_SOURCE_DIR}/deps
+    DOWNLOAD_NO_PROGRESS 1
+    DOWNLOAD_NAME V1.0.tar.gz
+    URL https://github.com/Shareong/paillier/archive/V1.0.tar.gz
+    URL_HASH SHA256=a5271ec419e93813163fac4dd38e052d850119ada8ca4427f9ff2816eca5e03e
+    BUILD_IN_SOURCE 1
+    SOURCE_SUBDIR ./homadd/
+    CMAKE_COMMAND ${CMAKE_COMMAND}
+    LOG_CONFIGURE 1
+    LOG_BUILD 1
+    LOG_INSTALL 1
+    BUILD_COMMAND make
+    INSTALL_COMMAND bash -c "cp *.${PAILLIER_LIB_SUFFIX} ${CMAKE_SOURCE_DIR}/deps/lib"
```

Comment:
I suggest don't use `bash -c cp`, because we can modify paillier

Reply:
done